src/openrct2/windows/ride.c

```diff
@@ -2487,7 +2487,7 @@ static rct_string_id window_ride_get_status_vehicle(rct_window *w, void *argumen
 			trackType == TRACK_ELEM_60_DEG_UP_TO_FLAT ||
 			trackType == TRACK_ELEM_DIAG_25_DEG_UP_TO_FLAT ||
 			trackType == TRACK_ELEM_DIAG_60_DEG_UP_TO_FLAT) {
-			if (track_type_is_invented(ride->type, TRACK_BLOCK_BRAKES) && vehicle->velocity == 0) {
+			if (track_piece_is_available_for_ride_type(ride->type, TRACK_BLOCK_BRAKES) && vehicle->velocity == 0) {
```

Comment:
I think it prevent 'Stopped by block brakes' from appearing if the vehicle has just stalled on one of these pieces on a non-block brake ride.